src/qgis_geonode/apiclient/apiv2.py

```diff
@@ -163,12 +163,27 @@ def get_geonode_resource(
         license_ = license_value["identifier"]
     else:
         license_ = license_value
+    default_style_dict = deserialized_resource.get("default_style", "")
+    default_style = models.BriefGeonodeStyle(
+        pk=default_style_dict.get("pk", ""),
+        name=default_style_dict.get("name", ""),
+        sld_url=default_style_dict.get("sld_url", "")
+    )
```

Comment:
This code is flawed. You try to get the default style and if not available it defaults to an empty string. Then you proceed to use the `dict.get()` method on it as if it is always a dict. If it is a string this will crash.

Lets assume there will always be a default style for all layers, as I can't think of a case where that would not be true.


```suggestion
    default_style = models.BriefGeonodeStyle(
        pk=deserialized_resource["default_style"]["pk"],
        name=deserialized_resource["default_style"]["name"],
        sld_url=deserialized_resource["default_style"]["sld_url"]
    )
```